db/1.9/modules/ROOT/pages/database-insert-update-delete.adoc

```diff
@@ -0,0 +1,218 @@
+    <db:sql>
+        INSERT INTO electronic(name, description, price, discount)
+        VALUES (:name, :description, :price, :discount)
+    </db:sql>
+    <db:input-parameters>#[{
+        name: 'Coffee Machine',
+        description: 'Model:XYZ99. Uses small size capsules.',
+        price: 120,
+        discount: 5}]
+    </db:input-parameters>
+</db:insert>
+----
+
+=== Configure the Auto Generate Keys Fields
+
+The Insert operation outputs a `Statement Result` object that contains two fields: an `affectedRows`
+integer that indicates how many rows were affected by the query, and a `generatedKeys` map
+that contains the autogenerated keys.
+
+In the following example, the output shows that the `generatedKeys` map is empty, although a row is successfully added to the table. The map is empty because this information is disabled by default to avoid overhead.
+
+[source,json,linenums]
+----
+{
+  "affectedRows": 1,
+  "generatedKeys": {
+
+  }
+}
+----
+
+
+To activate the information and generate keys, follow the next steps:
+
+ . In your Studio flow, select the *Insert* operation.
+ . In the *Advanced* tab, set the *Auto generate keys* field to `True`.
+
+The following screenshot shows the *Auto generate keys* field configuration:
+
+image::database-insert-example-3.png[Input Parameter configuration in Insert Operation]
+
+In the XML editor, the `<autoGenerateKeys>` configuration looks like this:
+
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true">
+...
+</db:insert>
+----
+
+If the statement generates multiple keys per affected row, you can specify which columns
+should be returned by setting a list that contains the desired column names, for example:
+
+. In your Studio flow, select the *Insert* operation.
+. In the *Advanced* tab, set the *Auto generate column names* field to `Expression or Bean reference` to set the field to a DataWeave list with a single string element.
+. Set the expression field to `['id']` to ensure that only the `id` column is returned.
+
+The following screenshot shows the *Auto generate column names* field configuration:
+
+image::database-insert-example-4.png[Auto generate column names configuration in Insert Operation]
+
+In the XML editor, the `<autoGeneratedKeysColumnNames>` configuration looks like this:
+
+[source,xml,linenums]
+----
+<db:insert config-ref="dbConfig" autoGenerateKeys="true" autoGeneratedKeysColumnNames="#[['id']]">
+    ...
+</db:insert>
+----
+
+== Configure the Update Operation in Studio
+
+Given the `electronic` table previously defined, to update data from the table, for example setting the discount to 10 percent for all items that have a price value above 100, configure the Update operation:
+
+. In your Studio flow, select the *Update* operation.
+. In the connector configuration screen, set the *SQL Query Text* field to the query to execute, for example: `"UPDATE electronic SET discount = :discount WHERE price > :price"`
```

Comment:
Did you intend for this all to be one sentence, on "one" line rather than setting off the syntax example?
If so:
```suggestion
. In the connector configuration screen, set the *SQL Query Text* field to the query to execute, for example, `"UPDATE electronic SET discount = :discount WHERE price > :price"`.
```